packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -271,15 +271,21 @@ export function FileUploader({
 
   // Tracker methods
 
+  const isValidExtension = (fileName: string, fileName2: string): boolean => {
+    const extension = fileName.split('.').pop();
+    const fileExtension = fileName2.split('.').pop();
+    return fileExtension === extension;
+  };
```

Comment:
I would either memoize this function with a useCallback, or better yet, move it outside the component in a util file, to prevent unnecessary re-renders from happenning.

Reply:
OK, I'll move it into the uploader utils.

Reply:
[30bd640](https://github.com/aws-amplify/amplify-ui/pull/3072/commits/30bd640172b8548e7bd2ad25a0bc47851f82bb50)